Alternative payments/Apple Pay Integration/IntegrateApplePay/Modules/ApplePay/JavaScript/ApplePay.View.js

```diff
@@ -0,0 +1,194 @@
+// @module DC.IntegrateApplePay.ApplePay
+define('DC.IntegrateApplePay.ApplePay.View', [
+	'dc_integrateapplepay_applepay.tpl',
+	'DC.IntegrateApplePay.ApplePay.Model',
+	'Backbone',
+	'LiveOrder.Model',
+	'jQuery'
+], function (
+	dc_integrateapplepay_applepay_tpl, ApplePayModel, Backbone, LiveOrderModel, jQuery
+) {
+	'use strict';
+
+	// @class DC.IntegrateApplePay.ApplePay.View @extends Backbone.View
+	return Backbone.View.extend({
+
+		template: dc_integrateapplepay_applepay_tpl,
+		initialize: function (options) {
+			this.container = options.container;
+			var checkout = this.container.getComponent("Checkout");
+			this.liveOrderModel = LiveOrderModel.getInstance();
+
+			var _this = this;
+			checkout.getCurrentStep().then(function (step) {
+				_this.stepInfo = step;
+			});
+			$.ajax({
+				url: "https://applepay.cdn-apple.com/jsapi/v1/apple-pay-sdk.js",
+				dataType: "script",
+				success: function () {
+					console.log("loaded");
+				}
+			});
+		},
+		events: {
+			'click #altapay-apple-pay-btn': 'applePayClicked'
+		},
+		applePayClicked: function () {
+			const self = this;
+			const request = {
+				"countryCode": this.container.getConfig().ApplePay.countryCode.toUpperCase() || "US",
+				"currencyCode": this.container.getConfig().ApplePay.currencyCode.toUpperCase() || "USD",
+				"merchantCapabilities": [
+					"supports3DS"
+				],
+				"supportedNetworks": [
+					"visa",
+					"masterCard"
+				],
+				"total": {
+					"label": this.container.getConfig().ApplePay.amountLabel || "AltaPay Apple Charge",
+					"type": "final",
+					"amount": this.liveOrderModel.get('summary').total
+				}
+			};
+
+			const session = new ApplePaySession(3, request);
+
+			session.onvalidatemerchant = async event => {
+
+				var settings = {
+					"validationUrl": event.validationURL
+				};
+
+				var applePayModel = new ApplePayModel();
+				applePayModel.fetch({
+					async: false,
+					data: settings
+				}).done((response) => {
+					if (response && response.response !== undefined) {
+						let parseJson = self.parseXmlToJson(response.response);
+
+						let altaPaySession = parseJson.APIResponse.Body.ApplePaySession.replace("<![CDATA[", "").replace("]]>", "");
+
+						if (parseJson && parseJson.APIResponse && parseJson.APIResponse.Body.Result.toLowerCase() === 'success') {
+							session.completeMerchantValidation(JSON.parse(altaPaySession));
+						} else {
+							console.log("response", "couldn't validate session");
+						}
+					}
+				}).fail((err) => {
+					console.log("error:", err);
```

Comment:
Don't you want to display in the website that there has been an error?

Reply:
Added alert dialog.